### PR TITLE
[SchemaRegistry] Update swagger and enable live testing

### PR DESCRIPTION
**Pull request by MiYanni**
- Updates to the latest swagger (which had some issues). I workaround these by using a local copy.
  - Will be adding AutoRest issues related to these problems
- Uses latest API version so test resources work appropriately
- Updates client code to remove hotfix from previous release
- Enables live tests to run for tests and samples
- Updates samples to work appropriately with live runs

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
LGTM, assuming the service has been rolled out to all clouds in the test matrix.   Are we sure about gov clouds and such?

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/src/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// Licensed under the MIT License.
+
+// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0100:ConfigureAwait(false) must be used.", Justification = "<Pending>", Scope = "member", Target = "~M:Azure.Data.SchemaRegistry.SchemaRestClient.GetByIdAsync(System.String,System.Threading.CancellationToken)~System.Threading.Tasks.Task{Azure.Core.ResponseWithHeaders{System.String,Azure.Data.SchemaRegistry.SchemaGetByIdHeaders}}")]
```
Comment (MiYanni) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/src/GlobalSuppressions.cs:11]: This was a workaround for this: https://github.com/Azure/autorest.csharp/pull/939

Once the generator is updated in this repo, this file can be deleted.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md [hunk 2]
```diff
@@ -42,13 +42,9 @@ The simpliest way is to use the [Azure portal][azure_portal] and navigate to you
...
-string endpoint = "<event_hubs_namespace_hostname>";
-var credentials = new ClientSecretCredential(
-    "<tenant_id>",
-    "<client_id>",
-    "<client_secret>"
-);
-var client = new SchemaRegistryClient(endpoint, credentials);
+// Create a new access Spark batch client using the default credential from Azure.Identity using environment variables previously set,
+// including AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, and AZURE_TENANT_ID.
+var client = new SchemaRegistryClient(endpoint: endpoint, credential: new DefaultAzureCredential());
```
Comment (jsquire) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:47]: Should we link over to the Identity ReadMe and let that be the authoritative voice for configuring your environment for the credential?   That may help keep this from getting stale in the face of any Identity changes in the future.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md [hunk 3]
```diff
@@ -78,8 +74,7 @@ The following shows examples of what is available through the SchemaRegistryClie
 Register a schema to be stored in the Azure Schema Registry.
 
 ```C# Snippet:SchemaRegistryRegisterSchema
-string schemaName = "<schema_name>";
-string groupName = "<schema_group_name>";
+string schemaName = "employeeSample";
```
Comment (jsquire) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:77]: Did you mean to remove the generic blurb here?

File: sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/README.md [hunk 4]
```diff
@@ -42,13 +42,9 @@ The simpliest way is to use the [Azure portal][azure_portal] and navigate to you
...
 
 ```C# Snippet:SchemaRegistryAvroCreateSchemaRegistryClient
-string endpoint = "<event_hubs_namespace_hostname>";
-var credentials = new ClientSecretCredential(
-    "<tenant_id>",
-    "<client_id>",
-    "<client_secret>"
-);
-var client = new SchemaRegistryClient(endpoint, credentials);
+// Create a new access Spark batch client using the default credential from Azure.Identity using environment variables previously set,
```
Comment (jsquire) [inline sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/README.md:45]: Should we link over to the Identity ReadMe and let that be the authoritative voice for configuring your environment for the credential? That may help keep this from getting stale in the face of any Identity changes in the future.

File: sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs [hunk 5]
```diff
@@ -11,54 +11,59 @@
...
 {
-    [Ignore("Only verifying that the sample builds")]
     public class Sample01_ReadmeSnippets : SamplesBase<SchemaRegistryClientTestEnvironment>
     {
-        [Ignore("Only verifying that the sample builds")]
-        [Test]
+#pragma warning disable IDE1006 // Naming Styles
+        private SchemaRegistryClient schemaRegistryClient;
+#pragma warning restore IDE1006 // Naming Styles
+        private byte[] _memoryStreamBytes;
```
Comment (jsquire) [inline sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs:19]: With the way that `_memoryStreamBytes` is used, it looks like you're relying on the tests running in a specific order and not being parallelized.    You may want to consider adding `NonParallelizable` and tagging the tests with `Order` to ensure deterministic execution.

**Review by jsquire — APPROVED** [review]
LGTM.

**Comment by jsquire** [comment]
I am still curious if we have confirmation that the service has rolled out to all the public and gov clouds at this point.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md (hunk 2 above)
Comment (MiYanni) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:47]: Not a bad idea. I copy/pasted this comment from the several other libraries that had the same thing. Also, just realized it mentions a Spark client. LOL. Forgot to actually read the comment.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md (hunk 3 above)
Comment (MiYanni) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:77]: I saw that other samples from other SDKs just make generic names for things that are irrelevant in actually running the sample code. Here, the name of the schema doesn't particularly matter.

File: sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs (hunk 5 above)
Comment (MiYanni) [reply sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs:19]: `NonParallelizable` is on the base sample class (part of the core test framework). I didn't know about `Order`. Instead, I just ran the tests within the other tests. I'll use `Order` instead.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md (hunk 2 above)
Comment (MiYanni) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:48]: Added.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md (hunk 3 above)
Comment (MiYanni) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/README.md:78]: Kept this in the sample.

File: sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/README.md (hunk 4 above)
Comment (MiYanni) [reply sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/README.md]: Added.

File: sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs (hunk 5 above)
Comment (MiYanni) [reply sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/Samples/Sample01_ReadmeSnippets.cs:19]: Added Order attributes.

**Comment by MiYanni** [comment]
@jsquire I have no idea. I was only given the green light to use the test resources template that JS used to do their live testing. So, I grabbed that and modified it. Then, I had to update to the new swagger to access the newer API version appropriately for testing. I don't have service details like that.

**Pull request merged**